src/models/dates/DateAsserter.ts

```diff
@@ -0,0 +1,84 @@
+import { DateType } from './DateFormat';
+import { DateComponents } from './DateSplitters';
+
+export const months: string[] = [
```

Comment:
You have done quite a lot of work to manage the date formats, was there no available library that could have done this for us?

Reply:
There is a good one called moment.js. I'll use that instead